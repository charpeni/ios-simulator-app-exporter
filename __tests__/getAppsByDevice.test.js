const getAppsByDevice = require('../getAppsByDevice');

it('should get devices correctly', () => {
  const appsByDevice = getAppsByDevice(`${__dirname}/Devices`);

  // We have to trim the absolute path as `homedir()` will be different for everyone.
  const trimPaths = appsByDevice.map(device => {
    if (device.apps.length > 0) {
      return {
        ...device,
        path: device.path.split('__tests__')[1],
        apps: device.apps.map(app => ({
          ...app,
          file: app.file.split('__tests__')[1],
        })),
      };
    }

    return {
      ...device,
      path: device.path.split('__tests__')[1],
    };
  });
  expect(trimPaths).toMatchSnapshot();
});
