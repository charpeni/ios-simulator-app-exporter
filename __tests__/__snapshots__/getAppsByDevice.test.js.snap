// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get devices correctly 1`] = `
Array [
  Object {
    "UDID": "148019A3-566F-4C21-A4FC-6FD75D9F051E",
    "apps": Array [
      Object {
        "CFBundleDisplayName": "pasta",
        "CFBundleIdentifier": "org.reactjs.native.example.pasta",
        "CFBundleShortVersionString": "3.0",
        "CFBundleVersion": "8",
        "file": "/Devices/148019A3-566F-4C21-A4FC-6FD75D9F051E/data/Containers/Bundle/Application/43FB53D5-6F42-4F38-A810-6C1A28B393E9/example.app",
      },
      Object {
        "CFBundleDisplayName": "example",
        "CFBundleIdentifier": "org.reactjs.native.example.example",
        "CFBundleShortVersionString": "1.0",
        "CFBundleVersion": "1",
        "file": "/Devices/148019A3-566F-4C21-A4FC-6FD75D9F051E/data/Containers/Bundle/Application/950806E1-3259-4089-8EF3-DF0941E3B5B7/example.app",
      },
    ],
    "deviceType": "com.apple.CoreSimulator.SimDeviceType.iPhone-7",
    "isDeleted": false,
    "name": "iPhone 7",
    "path": "/Devices/148019A3-566F-4C21-A4FC-6FD75D9F051E",
    "runtime": "com.apple.CoreSimulator.SimRuntime.iOS-12-1",
    "state": 3,
  },
  Object {
    "UDID": "1E756F87-FF14-45C0-9775-A4469F72599E",
    "apps": Array [
      Object {
        "CFBundleDisplayName": "example",
        "CFBundleIdentifier": "org.reactjs.native.example.example",
        "CFBundleShortVersionString": "1.0",
        "CFBundleVersion": "1",
        "file": "/Devices/1E756F87-FF14-45C0-9775-A4469F72599E/data/Containers/Bundle/Application/B3674201-2F90-4759-988E-CDB42181FD80/example.app",
      },
    ],
    "deviceType": "com.apple.CoreSimulator.SimDeviceType.iPad-Air",
    "isDeleted": false,
    "name": "iPad Air",
    "path": "/Devices/1E756F87-FF14-45C0-9775-A4469F72599E",
    "runtime": "com.apple.CoreSimulator.SimRuntime.iOS-9-3",
    "state": 3,
  },
  Object {
    "UDID": "7BBD33B7-63B0-4D96-9BE0-BDCA97FE89E0",
    "apps": Array [],
    "deviceType": "com.apple.CoreSimulator.SimDeviceType.iPhone-6s",
    "name": "iPhone 6s",
    "path": "/Devices/7BBD33B7-63B0-4D96-9BE0-BDCA97FE89E0",
    "runtime": "com.apple.CoreSimulator.SimRuntime.iOS-11-4",
    "state": 1,
  },
  Object {
    "UDID": "C321F874-FA51-4DC5-8B9A-2E6E095D8C90",
    "apps": Array [
      Object {
        "CFBundleDisplayName": "example",
        "CFBundleIdentifier": "org.reactjs.native.example.example",
        "CFBundleShortVersionString": "1.0",
        "CFBundleVersion": "1",
        "file": "/Devices/C321F874-FA51-4DC5-8B9A-2E6E095D8C90/data/Containers/Bundle/Application/4FEF6828-573D-4ACA-922D-85C1EEA7E509/example.app",
      },
    ],
    "deviceType": "com.apple.CoreSimulator.SimDeviceType.iPhone-XR",
    "isDeleted": false,
    "name": "iPhone XR",
    "path": "/Devices/C321F874-FA51-4DC5-8B9A-2E6E095D8C90",
    "runtime": "com.apple.CoreSimulator.SimRuntime.iOS-12-1",
    "state": 3,
  },
]
`;
